File: src/main/java/xyz/oribuin/lilori/commands/music/CmdLoop.java
package xyz.oribuin.lilori.commands.music;

import xyz.oribuin.lilori.managers.command.Command;
import xyz.oribuin.lilori.managers.command.CommandEvent;
import xyz.oribuin.lilori.managers.music.TrackManager;

import java.util.Collections;

public class CmdLoop extends Command {

    private boolean looping;

    public CmdLoop() {
        this.name = "Loop";
        this.aliases = Collections.emptyList();
        this.description = "Toggle song looping on/off";
    }

    @Override
    public void executeCommand(CommandEvent event) {
        if (event.getMember().getVoiceState() == null || !event.getMember().getVoiceState().inVoiceChannel()) {
            event.reply(event.getAuthor().getAsMention() + ", Could not loop since you are not in the voice channel");
            return;
        }

        if (this.isLooping()) {
            event.reply("Now ");
            this.setLooping(false);
        } else {
            event.reply("✅ Now Looping.");
            this.setLooping(true);
        }
    }

    private void setLooping(boolean looping) {
        this.looping = looping;
    }

    public boolean isLooping() {
        return looping;
    }
}

File: src/main/kotlin/xyz/oribuin/lilori/commands/global/music/CmdLoop.kt
package xyz.oribuin.lilori.commands.global.music

import xyz.oribuin.lilori.LilOri
import xyz.oribuin.lilori.handler.Category
import xyz.oribuin.lilori.handler.Command
import xyz.oribuin.lilori.handler.CommandEvent

class CmdLoop(bot: LilOri) : Command(bot) {
    var isLooping = false
        set


    init {
        name = "Loop"
        category = Category(Category.Type.MUSIC)
        aliases = emptyList()
        description = "Toggle song looping on/off"
        arguments = emptyList()
    }

    override fun executeCommand(event: CommandEvent) {
        if (event.member?.voiceState == null || event.member?.voiceState?.inVoiceChannel() == false) {
            event.reply(event.author.asMention + ", Could not loop since you are not in the voice channel")
            return
        }

        if (isLooping) {
            event.reply("✅ No Longer Looping. ")
            isLooping = false
        } else {
            event.reply("✅ Now Looping.")
            isLooping = true
        }
    }
}